lib/File/Ignore.pm6

```diff
@@ -55,7 +55,7 @@ class File::Ignore {
                 make '<' ~
                     ($<negate> ?? '-' !! '') ~
                     '[' ~
-                    $/.subst('\\', '\\\\', :g) ~
+                    $/.subst('\\', '\\\\', :g).subst('-', '..', :g) ~
```

Comment:
I think a `-` at the end of a charclass should be treated as a literal `-`, so we should perhaps fix that so we don't spit out a regex with a syntax error. Something like `.subst(/'-' )> ./, '..', :g)` would probably work. A test for that case would be good also.